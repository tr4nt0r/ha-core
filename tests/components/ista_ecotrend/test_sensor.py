"""Tests for the ista EcoTrend Sensors."""

from unittest.mock import MagicMock

from syrupy.assertion import SnapshotAssertion

from homeassistant.config_entries import ConfigEntryState
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from tests.common import MockConfigEntry


async def test_setup(
    hass: HomeAssistant,
    ista_config_entry: MockConfigEntry,
    mock_ista: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup of ista EcoTrend sensor platform."""

    ista_config_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(ista_config_entry.entry_id)
    await hass.async_block_till_done()

    assert ista_config_entry.state is ConfigEntryState.LOADED

    for entity in er.async_entries_for_config_entry(
        entity_registry, ista_config_entry.entry_id
    ):
        assert entity == snapshot
        assert hass.states.get(entity.entity_id) == snapshot
