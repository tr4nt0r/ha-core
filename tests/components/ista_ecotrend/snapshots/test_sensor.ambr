# serializer version: 1
# name: test_setup
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_heating',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 0,
      }),
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Heating',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HEATING: 'heating'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_heating',
    'unit_of_measurement': None,
  })
# ---
# name: test_setup.1
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Luxemburger Str. 1 Heating',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.luxemburger_str_1_heating',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '35',
  })
# ---
# name: test_setup.10
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': <RegistryEntryDisabler.INTEGRATION: 'integration'>,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_hot_water_cost',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Hot water cost',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HOT_WATER_COST: 'hot_water_cost'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_hot_water_cost',
    'unit_of_measurement': 'EUR',
  })
# ---
# name: test_setup.11
  None
# ---
# name: test_setup.12
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': <RegistryEntryDisabler.INTEGRATION: 'integration'>,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_water',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.WATER: 'water'>,
    'original_icon': None,
    'original_name': 'Water',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.WATER: 'water'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_water',
    'unit_of_measurement': <UnitOfVolume.CUBIC_METERS: 'm³'>,
  })
# ---
# name: test_setup.13
  None
# ---
# name: test_setup.14
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': <RegistryEntryDisabler.INTEGRATION: 'integration'>,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_water_cost',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Water cost',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.WATER_COST: 'water_cost'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_water_cost',
    'unit_of_measurement': 'EUR',
  })
# ---
# name: test_setup.15
  None
# ---
# name: test_setup.2
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_heating_energy',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.ENERGY: 'energy'>,
    'original_icon': None,
    'original_name': 'Heating energy',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HEATING_ENERGY: 'heating_energy'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_heating_energy',
    'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
  })
# ---
# name: test_setup.3
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'energy',
      'friendly_name': 'Luxemburger Str. 1 Heating energy',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.luxemburger_str_1_heating_energy',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '38.0',
  })
# ---
# name: test_setup.4
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': <RegistryEntryDisabler.INTEGRATION: 'integration'>,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_heating_cost',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.MONETARY: 'monetary'>,
    'original_icon': None,
    'original_name': 'Heating cost',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HEATING_COST: 'heating_cost'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_heating_cost',
    'unit_of_measurement': 'EUR',
  })
# ---
# name: test_setup.5
  None
# ---
# name: test_setup.6
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_hot_water',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.WATER: 'water'>,
    'original_icon': None,
    'original_name': 'Hot water',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HOT_WATER: 'hot_water'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_hot_water',
    'unit_of_measurement': <UnitOfVolume.CUBIC_METERS: 'm³'>,
  })
# ---
# name: test_setup.7
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'water',
      'friendly_name': 'Luxemburger Str. 1 Hot water',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': <UnitOfVolume.CUBIC_METERS: 'm³'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.luxemburger_str_1_hot_water',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '1.0',
  })
# ---
# name: test_setup.8
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.luxemburger_str_1_hot_water_energy',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.ENERGY: 'energy'>,
    'original_icon': None,
    'original_name': 'Hot water energy',
    'platform': 'ista_ecotrend',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': <IstaSensorEntity.HOT_WATER_ENERGY: 'hot_water_energy'>,
    'unique_id': '26e93f1a-c828-11ea-87d0-0242ac130003_hot_water_energy',
    'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
  })
# ---
# name: test_setup.9
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'energy',
      'friendly_name': 'Luxemburger Str. 1 Hot water energy',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.luxemburger_str_1_hot_water_energy',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '57.0',
  })
# ---
